src/main/java/io/r2dbc/proxy/core/MethodExecutionInfo.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.proxy.core;
+
+import java.lang.reflect.Method;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Hold method execution related information.
+ *
+ * @author Tadaya Tsuyukubo
+ */
+public class MethodExecutionInfo {
```

Comment:
Spring/Zipkin have various levels of exposure regarding this kind of object structures. E.g. `MethodExecutionInfo` (exposing read-only methods) and `MutableMethodExecutionInfo` so things get clear very quickly what the intent is.